src/transformers/adapters/model_mixin.py

```diff
@@ -274,6 +281,13 @@ def add_adapter(self, adapter_name: str, config=None, overwrite_ok: bool = False
         self.config.adapters.add(adapter_name, config=config)
         try:
             self.apply_to_adapter_layers(lambda i, layer: layer.add_adapter(adapter_name, i))
+            # PHM Layer
+            if config:
+                if (
+                    isinstance(config, ConfigUnion)
+                    and any("phm_layer" in config[i] and config[i]["phm_layer"] for i in range(len(config.configs)))
+                ) or (not isinstance(config, ConfigUnion) and "phm_layer" in config and config["phm_layer"]):
```

Comment:
I believe checking this here won't always work since `config` might be a string 🤔 Probably, we should retrieve the config from `self.config.adapters`

Reply:
```suggestion
            if self.config.adapters.match(adapter_name, AdapterConfig, location_key="phm_layer"):
```

This might actually work instead (haven't tried)